lib/registerTypes/RegisterTypes.h

```diff
@@ -201,6 +201,7 @@ void registerTypesMapObjects2(Serializer &s)
 	//s.template registerType<CObstacleInstance>();
 		s.template registerType<CObstacleInstance, MoatObstacle>();
 		s.template registerType<CObstacleInstance, SpellCreatedObstacle>();
+		s.template registerType<CObstacleInstance, BattlefieldLimiterObstacle>();

```

Comment:
you should register between registerTypesMapObjectTypes and registerTypesClientPacks1, may add new method for current additions and use it for new types until we decide to break save compatibility.